src/hooks/useFetch.tsx

```diff
@@ -0,0 +1,24 @@
+import {useEffect, useState} from "react";
+import {supabase} from "@/config/supabase";
+import {Agents} from "@/types";
+
+export const useFetch = () => {
```

Comment:
hook name is too global and you're just fetching the agents with this hook, but useFetch can be used for something else...